versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/util/SupplyOnce.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.common.util;
+
+import java.util.function.Supplier;
+
+/**
+ * This is an <em>internal</em> utility class, which provides an unsynchronized, not thread-safe
+ * {@link Supplier} that memoizes returned value but also a thrown exception.
+ */
+public final class SupplyOnce {
+
+  private SupplyOnce() {}
+
+  public static <T> Supplier<T> supplyOnce(Supplier<T> loader) {
```

Comment:
This is neat, but at the time time slightly confusing :) Makes me wonder (at call sites) what to expect on the second `.get()` call... Maybe simply `memoize(Supplier<T> loader)`?

Reply:
done